src/Nri/Ui/PremiumCheckbox/V1.elm

```diff
@@ -22,13 +21,15 @@ import Nri.Ui.Data.PremiumLevel as PremiumLevel exposing (PremiumLevel(..))
     If you get this message, you should show an `Nri.Ui.Premium.Model.view`
 
 -}
-type alias PremiumConfig msg =
+type alias PremiumConfig level msg =
     { label : String
     , id : String
     , selected : Checkbox.IsSelected
     , disabled : Bool
-    , teacherPremiumLevel : PremiumLevel
-    , contentPremiumLevel : PremiumLevel
+    , teacherPremiumLevel : level
+    , contentPremiumLevel : level
```

Comment:
💭 hmm,  if we're passing the levels just so the checkbox can pass them to functions that we are also passing, then why not just pass the bools `isFree : Bool, isAllowed : Bool` instead of passing the levels and the functions?